### PR TITLE
Support hierarchical commands

### PR DESCRIPTION
**Pull request by crowecawcaw**
## Overview

This PR eliminates fragile string parsing throughout the codebase and replaces it with robust metadata-based approaches. The changes make the code more maintainable, future-proof, and less prone to edge case failures.

### 🔧 **String Parsing Elimination**
- **Hierarchical tool detection**: Replace string splitting with stored metadata lookup
- **Child command lookup**: Store actual command references instead of trying string variations
- **Path parsing**: Store path components during creation instead of splitting dotted paths
- **Command name extraction**: Use stored child command names instead of string manipulation

### 🏷️ **MCP Command Detection**
- Replace fragile help text matching with attribute-based detection
- Mark MCP commands with `_is_mcp_command` attribute when created
- Works reliably with custom MCP command names
- Inline simple detection function for cleaner code

### 📁 **Professional Naming**
- Rename `test_context_issue.py` → `test_hierarchical_commands.py`
- Update function names to be descriptive rather than issue-specific
- Remove temporary/issue-specific references from comments and docstrings
- Use timeless, self-documenting names throughout

### Before (Fragile):
```python
# String parsing approach
if "_" in tool_name:
    parts = tool_name.split("_", 1)
    parent_name = parts[0]
    child_name = parts[1].replace("_", "-")

# Help text matching
if cmd_info.get("help") == "Start the MCP server...": 
    return True

# Path splitting
path_parts = original_path.split(".")
```

### After (Robust):
```python
# Metadata-based approach
parent_cmd = get_parent_command(tool_name)
child_name = get_child_command_name(tool_name)
child_cmd = get_child_command(tool_name)

# Attribute-based detection
if getattr(cmd, '_is_mcp_command', False):
    continue

# Stored components
path_components = get_command_path_components(tool_name)
```

## Benefits

- ✅ **More robust** - Won't break with unusual naming patterns or edge cases
- ✅ **Better maintainability** - Clear intent without fragile string parsing
- ✅ **Future-proof** - Uses stored metadata as source of truth
- ✅ **Professional** - Timeless naming conventions
- ✅ **Comprehensive testing** - All functionality thoroughly tested
- ✅ **No breaking changes** - Public API remains unchanged

## Testing

- All existing tests pass
- Added comprehensive test coverage for hierarchical commands
- Verified MCP command detection works with custom names
- Tested edge cases and parameter filtering

## Files Changed

- `click_mcp/decorator.py` - Add MCP command attribute marking
- `click_mcp/scanner.py` - Replace string parsing with metadata storage
- `click_mcp/server.py` - Use stored references instead of string matching
- `tests/` - Add comprehensive test coverage and professional naming

This PR significantly improves code quality and robustness while maintaining full backward compatibility.

### CODE REVIEW

File: tests/context_cli.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+This CLI has a parent command that sets ctx.obj and child commands
+that access that context, demonstrating hierarchical tool execution.
+"""
+
+import click
+from click_mcp import click_mcp
+
+
+@click_mcp(server_name="context-test-cli")
```
Comment (crowecawcaw) [inline tests/context_cli.py:13]: Extend this test app with multiple nested levels. Add parameters and context to every level. Then update the tests to verify deeply nested commands and context work.
